user-interface/src/main/java/life/qbic/datamanager/views/projects/project/experiments/experiment/components/ExperimentalGroupsDialog.java

```diff
@@ -174,13 +188,19 @@ private void refreshGroupEntries() {
   /**
    * Provides experimental groups defined by the user. Only group information in enabled components
    * is returned, as the list of existing groups cannot be changed in "add group" mode.
+   * <p>
+   * Note: {@link ExperimentalGroupInput} with no levels given will be ignored and do not result in
+   * an entry of the returned collection.
+   *
    * @return a collection of {@link ExperimentalGroupContent}, describing the group size (biological
    * replicates) and the variable level combination.
-   * @since 1.0.0
+   * @since 1.10.0
    */
   public Collection<ExperimentalGroupContent> experimentalGroups() {
     return this.experimentalGroupsCollection.getChildren()
         .filter(component -> component.getClass().equals(ExperimentalGroupInput.class))
+        // the order of the filter statement matters, since we do not want to throw an exception due to type mismatch
+        .filter(groupInput -> !((ExperimentalGroupInput) groupInput).getCondition().isEmpty())
         .map(experimentalGroupEntry -> convert((ExperimentalGroupInput) experimentalGroupEntry))
         .toList();
```

Comment:
Please think about separating the class casting and the parameter operations.
```suggestion
        .map(groupInput -> (ExperimentalGroupInput) groupInput)
        .filter(groupInput -> !groupInput.getCondition().isEmpty())
        .map(experimentalGroupEntry -> convert(experimentalGroupEntry))
        .toList();
```

Reply:
but the stream content has already been filtered correctly for the class types in question. I dont see the issue with that expression. 
Maybe you can explain it to me.

Reply:
We filtered for the correct class, however the compiler does not understand that the objects in the stream are now of class `ExperimentalGroupInput` and assumes the elements of the stream to be of `Component` class. Thus if we want to perform some action on an experimental group input, we need to cast to it. We can of course keep the `Component` stream and cast in each map, filter, or other call and only cast in the `toList` collector. Easier would be to cast the elements of the stream thus telling the compiler explicitly that we now have a stream of `ExperimentalGroupInput`s.